Automattic/newspack-plugin
# feat(ia): newspack \ settings boilerplate and general feature refactor/restructure

### All Submissions:


### Changes proposed in this Pull Request:

NB: The vast majority of the diff is from moving files to new locations.

* Newspack / Settings page & tabs boilerplate.
* Replaced `undefined` default assignment for Wizard component props with JSDoc comment - needed to appease TS when working with pure js components.
* Added `exact` prop to Wizard props (passed to `<Route exact />` component) - `exact` prop is required for `/` path.
* Standardized server defined local json (variables/properties) i.e. `wp_localize_script`, from snake_case to camelCase.
* Restructure directories to accommodate new I.A. in a readable and intuitive way.

<details open>
<summary>Current structure:</summary>

```md
dashboard
 ├ components
 │  ├ brand-header.tsx
 │  ├ icons
 │  │  ├ ad.tsx
 │  │  ├ block-post-date.tsx
 │  │  ├ dashboard.tsx
 │  │  ├ gift.tsx
 │  │  ├ index.ts
 │  │  ├ mail.tsx
 │  │  ├ position-center-center.tsx
 │  │  ├ post.tsx
 │  ├ quick-actions.tsx
 │  ├ sections.tsx
 │  ├ site-statuses
 │  │  ├ index.scss
 │  │  ├ index.tsx
 │  │  ├ site-status-modal.tsx
 │  │  ├ site-status.tsx
 ├ index.tsx
 ├ style.scss
 ├ types
 │  ├ index.d.ts
 │  ├ site-statuses.d.ts
 ├ views
 │  └ dashboardCard.js
```
</details>

<details>
<summary>New structure:</summary>

```md
newspack
 ├ components
 │  ├ brand-header.tsx
 │  ├ icons
 │  │  ├ ad.tsx
 │  │  ├ block-post-date.tsx
 │  │  ├ dashboard.tsx
 │  │  ├ gift.tsx
 │  │  ├ index.ts
 │  │  ├ mail.tsx
 │  │  ├ position-center-center.tsx
 │  │  ├ post.tsx
 │  ├ quick-actions.tsx
 │  ├ site-statuses
 │  │  ├ index.scss
 │  │  ├ index.tsx
 │  │  ├ site-status-modal.tsx
 │  │  ├ site-status.tsx
 ├ index.tsx
 ├ types
 │  ├ index.d.ts
 │  ├ site-statuses.d.ts
 ├ views
 │  ├ dashboard
 │  │  ├ index.tsx
 │  │  ├ sections.tsx
 │  │  ├ style.scss
 │  ├ settings
 │  │  ├ index.tsx
 │  │  ├ sections.tsx
 │  │  ├ style.scss
 │  │  ├ tabs
 │  │  └ types.d.ts
```
</details>

The main idea is to house each group of admin pages within a directory. Example: the structure within `assets/wizards/newspack` is intended to mimic the new menu configuration. This was motivated by the need to create the new Newspack / Settings page, however there was already a 'wizard' labelled "settings". I wasn't able to identify exact usage (or if used at all),  before continuing it became clear that a small refactor in this area was necessary. 

### How to test the changes in this Pull Request:

1. Checkout this branch and build i.e. `git checkout origin/feat/ia-settings && npm run build`
2. Navigate to Newspack / Settings - wp-admin/admin.php?page=newspack-settings#/
3. Observe you see the following...
![Screenshot 2024-04-12 at 09 55 08](https://github.com/Automattic/newspack-plugin/assets/3676975/893f4fe0-ee3a-4a3d-a7b8-f1b65f7aaf8f)
4. Ensure Newspack / Dashboard loads correctly too.

### Other information:
